pkg/core/storage/memory_store_test.go

```diff
@@ -75,3 +75,46 @@ func TestMemoryStore_Seek(t *testing.T) {
 		assert.Equal(t, value, v)
 	})
 }
+
+func TestMemoryStorePersist(t *testing.T) {
```

Comment:
nit: a very good candidate for parameterized test:

```
testCases := []struct { 
        key     string 
        value   string
    }{ 
        "key", "value"
    } 
 
    for _, tc := range testCases { 
        t.Run(tc.Name, func(t *testing.T) { 

...
```

Reply:
I'm not sure that's the case, these are nice when you've got clear input and clear output data (like pure function), but here we're dealing with the state of two structures and the pattern is not that clear. But at the same time, I do agree that in the current shape it's write-only code and it's not easy to follow, so I'll add some comments there.

Reply:
ok. probably we can get back to that later and think about some ways for improvement. Don't think that it's critical thing as of now